app/channels/editor_channel.rb

```diff
@@ -0,0 +1,59 @@
+# The Editor Channel is opened when a user navigates to the `Edit` page of an
+# Evidence, Issue, Note, or Card. The channel is closed when the user navigates
+# away from the `edit` page. When open the channel receives updates from the
+# editor and saves them as new revisions of the resource.
+class EditorChannel < ApplicationCable::Channel
+  include ProjectScopedChannels
+  include PaperTrailActivity
+
+  attr_accessor :resource
+
+  AUTOSAVE_EVENT = 'auto-save'.freeze
+
+  def subscribed
+    reject and return unless find_resource(params)
+
+    stream_for [current_user, current_project, resource]
+  end
+
+  def save(params)
+    resource.paper_trail_event = AUTOSAVE_EVENT
+
+    if resource.update_attributes parsed_params(params['data'])
+      track_activity(resource, AUTOSAVE_EVENT)
+
+      self.class.broadcast_to([current_user, current_project, resource], resource.reload.updated_at.to_i)
+    end
+  end
+
+  private
+
+  def find_resource(params)
+    return unless %w[evidence issue note card].include? params['resource_type']
+
+    if params['resource_type'] == 'card'
+      authorized_list_ids = List.where(board_id: current_project.boards.select(:id)).select(:id)
+      @resource ||= Card.find_by(id: params['resource_id'], list_id: authorized_list_ids)
+    else
+      @resource ||= current_project.send(params['resource_type'].pluralize).find_by id: params['resource_id']
```

Comment:
Doesn't look safe calling `send` with `params`. Can we use `parsed_params` here?

Reply:
I'm going to say it is safe. On line [#32](https://github.com/dradis/dradis-ce/pull/656/files#diff-3c89c0c2231ade55a853b21327ca6785R32) I do a check of approved possible values. We only let in what we want to let in.